.devcontainer/devcontainer.json

```diff
@@ -22,19 +20,22 @@
 	},
 
 	// Use 'postCreateCommand' to run commands after the container is created.
-	// "postCreateCommand": "uname -a",
-
-	// Configure tool-specific properties.
-	// "customizations": {},
+	// Here we run `colcon build` to build the workspace initially, (to speed up later builds), and we keep going if it fails for some packages to get as much compiled as possible
+	"postCreateCommand": ".devcontainer/post-create.sh",
 
 	// Uncomment to connect as root instead. More info: https://aka.ms/dev-containers-non-root.
 	"remoteUser": "ros",
+	"updateRemoteUserUID": true, // ensure internal user has the same UID as the host user and update file permissions
 	"customizations": {
 		"vscode": {
 			"extensions": [
 				"ms-python.python",
 				"GitHub.vscode-pull-request-github",
-				"ms-vscode.cpptools"
+				"ms-vscode.cpptools",
+				"JaehyunShim.vscode-ros2",
+				"nonanonno.vscode-ros2",
+				"deitry.colcon-helper",
+				"github.vscode-github-actions"
```

Comment:
I added a number of useful extensions in the devcontainer, feel free to change if you think it's overkill

Reply:
I can comment directly on code!! how is that working ...

Reply:
This devcontainer currently running on Coltrane :D via vpn

Reply:
So this `JaehyunShim.vscode-ros2` is mainly for providing code patterns for ROS2? and this `nonanonno.vscode-ros2` for syntax highlight? I believe they are very useful to have. Will keep them

Reply:
> I can comment directly on code!! how is that working ...

The `GitHub.vscode-pull-request-github` provides that functionality